### PR TITLE
[ONNX] Log instead of crash when 'tabulate' is not installed

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #105862
* __->__ #106228

### CODE REVIEW

File: torch/onnx/_internal/fx/_pass.py [hunk 1]
```diff
@@ -214,21 +218,26 @@ def run(self, *args, **kwargs) -> torch.fx.GraphModule:
         )
         # Gather graph information before transform.
         old_readable_graph = self.module.print_readable(print_output=False)
-        old_tabular = fx_graph_tabular(self.module.graph)
+        old_tabular = maybe_fx_graph_tabular(self.module.graph)
```
Comment (justinchuby) [inline torch/onnx/_internal/fx/_pass.py:221]: I think `fx_graph_tabular` returning None is clear and ok. For example, `dict.get()` doesn't have to be `dict.maybe_get()`.

I view a `maybe` function with the meaning of "do this, but if you can't do nothing and don't fail." Bottomline I would add it when it increases clarity. Here I think the original name w/ the type annotation is clear.

**Review by justinchuby — APPROVED** [review]
LGTM w/ naming suggestions

**Review by thiagocrepaldi — COMMENTED** [review]
What is the context to allow this?

Although `tabulate` may not be strictly required when the model exports successfully, on a failure scenario on the other hand, it is preferable to ask users to install `tabulate` than allow them to open a bug without the tabular graph in the bug description, for example.

I wonder whether the user will guess on its own that they must install `tabulate` before filing a bug or when debugging on their own. 

Being debug-ready is important for self-service and reporting issues.

Have you considered adding

```python
    try:
        import tabulate
    except ImportError as e:
        raise missing_package("tabulate", e) from e
```
to `[_assert_dependencies](https://github.com/pytorch/pytorch/blob/main/torch/onnx/_internal/exporter.py#L747) which is executed at the `dynamo_export`?

**Comment by BowenBao** [comment]
`print_tabular` is always called to form diagnostics for passes. All the context is described in the title. Not crashing due to non-export-related reason is more important than this section of diagnostic info. Perhaps one can either make `tabulate` a torch dependency, or rewrite `print_tabular` to not depend on `tabulate`.

File: torch/onnx/_internal/fx/_pass.py (hunk 1 above)
Comment (BowenBao) [reply torch/onnx/_internal/fx/_pass.py:221]: I'm not certain of this. In my mind `fx_graph_tabular` is `str` typed. `None` is not valid output. So `maybe` prefix hints why potentially None can be returned.

**Pull request closed without merging**